TFNetworkLayer.py

```diff
@@ -5677,6 +5677,12 @@ def init(self, output, output_with_activation=None, target=None, layer=None):
         else:
           assert not target.beam_size
       self.output = output
+      if self.output.is_batch_feature_major:
+        self.output = self.output.copy_with_feature_dim_axis(-1)
+        if output_with_activation:
+          from TFUtil import move_axis
+          output_with_activation = OutputWithActivation(x=move_axis(output_with_activation.x, 1, -1),
+                                                        act_func=output_with_activation.act_func)
```

Comment:
This is slightly inconsistent (you operate on `self.output`, i.e. the attribute, but on `output_with_activation`, i.e. not the attribute but the local var). Why don't you make it consistent? I.e. just do it in both cases on the local var (`output` and `output_with_activation`), before the assignment `self.output = output`.

Also, instead of the check `output.is_batch_feature_major`, you should check `output.feature_dim_axis is not None and output.feature_dim_axis != output.batch_ndim - 1`.